components/scream/src/control/atmosphere_driver.cpp

```diff
@@ -296,29 +306,29 @@ initialize_fields (const util::TimeStamp& t0)
           "       Ref grid name:    " + ref_grid_name + "\n"
           "       Input field grid: " + req.fid.get_grid_name() + "\n");
 
-      ic_fields.set(ekat::strint("field",ifield+1),name); 
+      ic_fields_names.push_back(name);
       ++ifield;
 
     }
     f.get_header().get_tracking().update_time_stamp(t0);
   }
+  ic_reader_params.set("FIELDS",ic_fields_names);
 
   // Check the status of lat and lon values on reference grid:
   // This option allows the user to set lat or lon in their own
   // test or run setup code rather than by file.
   bool skip_init_lat = true;
   bool skip_init_lon = true;
   if (ic_pl.isParameter("skip_init_lat")) {
-    skip_init_lat     = ic_pl.get<bool>("skip_init_lat");
+    skip_init_lat = ic_pl.get<bool>("skip_init_lat");
```

Comment:
Is `skip_init_{lat/lon}` a legacy name? It's less confusing to use terms that indicate things should be done rather than skipped. Skipping things implies assumptions that everyone knows. But if this name is historical, I guess we're stuck with it.

Reply:
I agree, I'll swap that around.